miio/push_server/server.py

```diff
@@ -90,6 +94,13 @@ async def stop(self):
         self._listen_couroutine = None
         self._loop = None
 
+    def add_method(self, name: str, response: Union[Dict, Callable]):
+        """Add a method to server.
+
+        The response can be either a callable or a dictionary to send back as response.
+        """
+        self._methods[name] = response
+
     def register_miio_device(self, device: Device, callback: PushServerCallback):
```

Comment:
I would add a check here:
```
if self._device_ip is None:
   warning
   return
```

To make sure this is a proper server that can work with real devices

Reply:
I'm not sure I'm following what you mean, the `device_ip` is checked right below?